From 872038799ed19500a09391cd2225d5b8fd2752a6 Mon Sep 17 00:00:00 2001
From: Evan Chen <evanxinchen@google.com>
Date: Tue, 23 Sep 2025 18:17:07 +0000
Subject: [PATCH] Validate displayName for AssociationRequest

validate the length of displayName in the constructor, setDisplayName
and the builder

Test: cts
Flag: EXEMPT BUGFIX
Bug: 443742829
Change-Id: I41dc3804c35328a5968716f502c4320a58b75158
---
 .../java/android/companion/AssociationRequest.java | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/core/java/android/companion/AssociationRequest.java b/core/java/android/companion/AssociationRequest.java
index 10d47559cf73..c27893d933bb 100644
--- a/core/java/android/companion/AssociationRequest.java
+++ b/core/java/android/companion/AssociationRequest.java
@@ -261,6 +261,7 @@ public final class AssociationRequest implements Parcelable {
             @Nullable CharSequence displayName,
             boolean selfManaged,
             boolean forceConfirmation) {
+        validateDisplayName(displayName);
         mSingleDevice = singleDevice;
         mDeviceFilters = requireNonNull(deviceFilters);
         mDeviceProfile = deviceProfile;
@@ -342,6 +343,7 @@ public final class AssociationRequest implements Parcelable {
 
     /** @hide */
     public void setDisplayName(CharSequence displayName) {
+        validateDisplayName(displayName);
         mDisplayName = displayName;
     }
 
@@ -421,10 +423,7 @@ public final class AssociationRequest implements Parcelable {
         public Builder setDisplayName(@NonNull CharSequence displayName) {
             checkNotUsed();
             mDisplayName = requireNonNull(displayName);
-            if (displayName.length() > DISPLAY_NAME_LENGTH_LIMIT) {
-                throw new IllegalArgumentException("Length of the display name must be at most "
-                        + DISPLAY_NAME_LENGTH_LIMIT + " characters");
-            }
+            validateDisplayName(displayName);
 
             return this;
         }
@@ -672,4 +671,11 @@ public final class AssociationRequest implements Parcelable {
             return new AssociationRequest(in);
         }
     };
+
+    private static void validateDisplayName(@Nullable CharSequence displayName) {
+        if (displayName != null && displayName.length() > DISPLAY_NAME_LENGTH_LIMIT) {
+            throw new IllegalArgumentException("Length of the display name must be at most "
+                    + DISPLAY_NAME_LENGTH_LIMIT + " characters");
+        }
+    }
 }
-- 
2.52.0.rc1.455.g30608eb744-goog

